integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ETagSupportTestCase.java

```diff
@@ -0,0 +1,141 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import org.wso2.carbon.automation.extensions.servers.httpserver.SimpleHttpClient;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ETagSupportTestCase extends ESBIntegrationTest {
+
+    //TODO: Enable tests after fixing fixing SimpleHttpClient issue TA-945
+
+    private String epr;
+    private SimpleHttpClient httpClient;
+    private Map<String, String> headers;
+    private Header eTagHeader;
+
+    private String payload;
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+
+        super.init();
+        httpClient = new SimpleHttpClient();
+        epr = getProxyServiceURLHttp("ETagTestProxy");
+
+        // Setting initial http headers
+        headers = new HashMap<String, String>();
+        headers.put("action", "urn:echoString");
+        headers.put("Content-Type", "application/soap+xml");
+        headers.put("charset", "UTF-8");
+
+        payload = "<soapenv:Envelope xmlns:soapenv=\"http://www.w3.org/2003/05/soap-envelope\">" +
+                  "<soapenv:Body>" +
+                  "<p:echoInt xmlns:p=\"http://echo.services.core.carbon.wso2.org\">" +
+                  "<in>1</in>" +
+                  "</p:echoInt>" +
+                  "</soapenv:Body>" +
+                  "</soapenv:Envelope>";
+
+    }
+
+    @Test(groups = "wso2.esb", description = "ETag enable test for a pass-through proxy", enabled = true)
+    public void testEnableETag() throws Exception {
+
+        HttpResponse response = httpClient.doPost(epr, headers, payload, "text/xml");
+
+        eTagHeader = response.getFirstHeader("ETag");
+        Assert.assertNotNull(eTagHeader, "ETag header is not present in the response");
+
+        int statusCode = response.getStatusLine().getStatusCode();
+        Assert.assertEquals(statusCode, 200, "Incorrect status code. Expected: 200 Found: " + statusCode);
+        Assert.assertNotNull(httpClient.getResponsePayload(response), "Payload should not empty");
+    }
+
+    @Test(groups = "wso2.esb", description = "If-None-Match header test for cache hit", enabled = false)
+    public void testIfNoneMatchHeaderCacheHit() throws Exception {
+        headers.put("If-None-Match", "\"" + eTagHeader.getValue() + "\"");
```

Comment:
For this test to run, the eTagHeader should be added by running, 'testEnableETag', right? Shall we add a dependancy on the prior test method?